### PR TITLE
allow the `GeoView` to be set as a `QObject`

### PR DESCRIPTION
**Pull request by lsmallwood**
@michael-tims, @JamesMBallard please review the changes to allow the `GeoView` to be supplied to the CoordinateController as a property.

### CODE REVIEW

File: Plugin/CppApi/source/CoordinateConversion/CoordinateConversionController.cpp [hunk 1]
```diff
@@ -259,12 +264,18 @@ bool CoordinateConversionController::isFormat(CoordinateConversionOptions *optio
...
+{
+  if (setGeoViewInternal(dynamic_cast<GeoView*>(geoView)))
+    connect(geoView, SIGNAL(mouseClicked(QMouseEvent&)), this, SLOT(onMouseClicked(QMouseEvent&)));
+}
+
 /*!
   \property CoordinateConversionController::results
   \brief The conversion results as a list model.
-
+
```
Comment (JamesMBallard) [inline Plugin/CppApi/source/CoordinateConversion/CoordinateConversionController.cpp]: Whitespace added.

File: Plugin/CppApi/source/CoordinateConversion/CoordinateConversionController.cpp [hunk 2]
```diff
@@ -259,12 +264,18 @@ bool CoordinateConversionController::isFormat(CoordinateConversionOptions *optio
...
+}
+
 /*!
   \property CoordinateConversionController::results
   \brief The conversion results as a list model.
-
+  
   The results are automatically updated as conversions are run.
-
+
```
Comment (JamesMBallard) [inline Plugin/CppApi/source/CoordinateConversion/CoordinateConversionController.cpp]: Whitespace added.

File: Plugin/CppApi/source/CoordinateConversion/CoordinateConversionController.cpp [hunk 3]
```diff
@@ -346,10 +368,15 @@ void CoordinateConversionController::setCaptureMode(bool captureMode)
 
   If the tool is active and in \l captureMode, this will be used as the input for conversions.
  */
-void CoordinateConversionController::onMouseClicked(const Point& clickedPoint)
+void CoordinateConversionController::onMouseClicked(QMouseEvent& mouse)
```
Comment (michael-tims) [inline Plugin/CppApi/source/CoordinateConversion/CoordinateConversionController.cpp]: Typically we'd call the parameter `event` or `mouseEvent` for QMouseEvent

**Comment by lsmallwood** [comment]
Changes made - merging

**Pull request merged**